docs/src/integrations/iceberg.md

```diff
@@ -210,13 +210,18 @@ The Iceberg Catalog supports all standard Iceberg table operations:
 
 ### Version Control Features
 
-The Iceberg Catalog integrates with lakeFS's version control system, treating each table change as a commit. 
-This provides a complete history of table modifications and enables branching and merging workflows.
+The Iceberg Catalog integrates with lakeFS's version control system. Table and namespace changes made through the 
+Iceberg API are staged on the target lakeFS reference, and may be committed to lakeFS to record these staged changes. 
+This provides a complete history of table modifications and enables lakeFS branching and merging workflows.
 
-#### Catalog Changes as Commits
+#### Catalog Changes in lakeFS
 
-Each modification to a table (schema changes, data updates, etc.) creates a new commit in lakeFS. 
-Creating or deleting a namespace or a table results in a lakeFS commit on the relevant branch, as well as table data updates ("Iceberg table commit").
+Creating, dropping, or committing changes to a table results in staging changes to a (table) lakeFS object, 
+and creating or dropping a namespace results in staging changes to a (namespace) lakeFS object.
```

Comment:
feel like we 2x "staging changes" - we can say it is that the iceberg changes will be reflected as staging changes on the lakeFS table object.